spec/solver_spec.rb

```diff
@@ -0,0 +1,37 @@
+require './solver'
+
+describe Solver do
+  describe '#factorial' do
+    it 'should return 1 for 0' do
+      Solver.new.factorial(0).should == 1
+    end
+    it 'should raise an error for negative numbers' do
+      expect { Solver.new.factorial(-1) }.to raise_error(RuntimeError)
+    end
+    it 'should return 120 for 5' do
+      Solver.new.factorial(5).should == 120
+    end
+  end
+  describe '#reverse' do
+    it 'should return olleh for hello' do
+      Solver.new.reverse('hello').should == 'olleh'
+    end
+  end
+  describe '#fizzbuzz' do
+    it 'should return fizz' do
+      Solver.new.fizzbuzz(3).should == 'fizz'
+    end
+
+    it 'should return buzz' do
+      Solver.new.fizzbuzz(10).should == 'buzz'
+    end
+
+    it 'should return fizzbuzz' do
+      Solver.new.fizzbuzz(30).should == 'fizzbuzz'
+    end
+
+    it 'should return number' do
+      Solver.new.fizzbuzz(7).should == (7)
```

Comment:
- [ ] You are required to return the string version of the number if it is not divisible by either 3 or 5. So in this case the result should be '7', not 7.

Reply:
@TedLivist Thanks for feedback.